csv/spec.go

```diff
@@ -0,0 +1,22 @@
+package csv
+
+import "fmt"
+
+//nolint:revive
+type CSVSpec struct {
```

Comment:
```suggestion
type Spec struct {
```

Ah, I didn't realize these can go in package called `csv`; in that case let's call it `Spec` so it will get called as `csv.Spec`

Reply:
If we do that then we will need some type functionality like an alias to be able to embed them in the same struct as they have the same name...

Reply:
Had a conversation with @hermanschaaf about this... I was hesitant to use the generic approach as there is no need to be able to handle arbitrary yaml/ inputs in this spec, but turns out we need this approach because otherwise there is no way of handling naming duplications between various file format configurations...